module_utils/network_lsr/ethtool.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+import array
+import binascii
+import struct
+import fcntl
+import socket
+
+
+ETHTOOL_GPERMADDR = 0x00000020
+SIOCETHTOOL = 0x8946
+
+
+def get_perm_addr(ifname):
+    """
+    Return the Permanent address value for the specified interface using the
+    ETHTOOL_GPERMADDR ioctl command. The address is returned as string in
+    python2 so it is needed to convert it to a bytearray and convert it again
+    to a hex string again.
+
+    Please for further documentation, see:
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L734
+    https://github.com/torvalds/linux/blob/master/include/uapi/linux/ethtool.h#L1388
+    https://git.kernel.org/pub/scm/network/ethtool/ethtool.git/tree/ethtool.c#n4172
+    """
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sockfd = sock.fileno()
+        ifname = ifname.encode("utf-8")
+        ecmd = array.array("B", struct.pack("II6s", ETHTOOL_GPERMADDR, 6, b"\x00" * 6))
+        ifreq = struct.pack("16sP", ifname, ecmd.buffer_info()[0])
+
+        fcntl.ioctl(sockfd, SIOCETHTOOL, ifreq)
+        res = ecmd.tostring()
+        _, _, perm_addr = struct.unpack("II6s", res)
+        if hasattr(perm_addr, "hex"):
+            perm_addr = perm_addr.hex()
+        else:
+            perm_addr = binascii.hexlify(perm_addr)
+            if not perm_addr:
```

Comment:
Why is this if here? When does this trigger?

Reply:
When the mac address is 00:00:00:00:00:00 on python2.6, there is no "tobytes" method so we are getting a string instead. When transforming it to string it is an empty string which is wrong. There is no other solution for this, or at least, I was not able to find it.

Reply:
I do not follow, when is `not perm_addr` True?

Reply:
Whops. You are right. I think I missed something then. Thanks for pointing this out. :-)